### PR TITLE
Problem 2

### PR DESCRIPTION
**Pull request by jallen3095**
Code for problem 2.

### CODE REVIEW

File: system/stackdepth.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	unsigned long *top_esp;
+	unsigned long *top_ebp;
+
+	unsigned long *sp;
+	unsigned long *fp;
+
+    int count = 0;
+
+	asm("movl %%esp, %0" : "=r" (top_esp));
```
Comment (npalacio) [inline system/stackdepth.c:25]: What do these lines do?

Comment (jallen3095) [reply system/stackdepth.c:25]: Those lines read the values of the "esp" and "ebp" registers and save them into local variables. I used the inline assembly language function as specified by the assignment instructions:

"To access the last stack frame pushed onto the run‐time stack, use the
in‐line assembly function asm() to read the values of %esp and %ebp and save them
into local variables".

Comment (SethRedwine) [reply system/stackdepth.c:25]: I believe him

**Comment by SethRedwine** [comment]
Yeet

**Review by SethRedwine — APPROVED** [review]
Looks like C

**Comment by SethRedwine** [comment]
That first button was wrong. 

Yeet2.0

**Comment by SethRedwine** [comment]
Apparently it takes more buttons than I am willing to read here, guys.

**Pull request merged**